JenkinsFile

```diff
@@ -1,18 +1,222 @@
+        -DartifactId=${appBaseName} \
+        -Dversion=${appvers}
+      """
+    }
+  }
+
+  stage('Scans') {
+    withCredentials([[$class: "StringBinding", credentialsId: "${env.THREADFIX_API_KEY}", variable: "THREADFIX_KEY"]]) {
+      // Dependency Checker
+      // sh "/jslave/dependency-check/dependency-check/bin/dependency-check.sh --version"
+      // sh '/jslave/dependency-check/dependency-check/bin/dependency-check.sh --project "${appBaseName}" --scan "." --format "XML" --enableExperimental --disableBundleAudit'
+      // sh "/bin/curl -v --insecure -H 'Accept: application/json' -X POST --form file=@dependency-check-report.xml ${env.THREADFIX_URL}/rest/applications/${env.THREADFIX_ID}/upload?apiKey=${THREADFIX_KEY}"
+      // Fortify
+      // sh "/opt/hp_fortify_sca/bin/sourceanalyzer -b ${env.BUILD_NUMBER} ./{*.py,**/*.py}"
+      // sh "/opt/hp_fortify_sca/bin/sourceanalyzer -b ${env.BUILD_NUMBER}  -scan -Xmx1G -f fortifyResults-${env.BUILD_NUMBER}.fpr"
+      // sh "/bin/curl -v --insecure -H 'Accept: application/json' -X POST --form file=@fortifyResults-${env.BUILD_NUMBER}.fpr ${env.THREADFIX_URL}/rest/applications/${env.THREADFIX_ID}/upload?apiKey=${THREADFIX_KEY}"
+    }
+
+    // TODO Put back in Sonar
+    /*
+    withCredentials([string(credentialsId: "${env.SONAR_TOKEN}", variable: 'USERNAME')]) {
+      withEnv(["HOME=${root}"]) {
+        def projectId = "venicegeo:beachfront:${appBaseName}"
+        sh "${mvn}/bin/mvn sonar:sonar -Dmaven.repo.local=${root} -Dsonar.host.url=${env.SONAR_URL} -Dsonar.projectKey=${projectId} -Dsonar.projectName=${projectId} -Dsonar.login=${USERNAME}"// -P nga"
+      }
+    }*/
+  }
+
+  stage("Security Scan Pass/Fail") {
+    // Query threadfix and block if any highs/crits are found
+    // TODO: Put back in
+    /*sh "mkdir -p sspf-pkg"
+    dir("./sspf-pkg") {
+      withCredentials([[$class: 'StringBinding', credentialsId: "${env.THREADFIX_API_KEY}", variable: 'THREADFIX_KEY']]) {
+        sh "curl -L ${env.SSPF_PACKAGE} -o sspf.zip"
+        sh "unzip sspf.zip"
+        dir("./sspf-master/sspf") {
+          sh "/bin/curl --fail --silent ${env.THREADFIX_URL}/rest/applications/${THREADFIX_ID}?apiKey=${THREADFIX_KEY} | ./sspf.py"
+        }
+      }
+    }*/
+  }
+
+  stage ('Int Deploy') {
+    if(!fileExists('.cf')) {
+      sh "mkdir -p .cf"
+    }
+    withEnv(['CF_HOME=.cf']) {
+      withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: "${env.PCF_CREDS}", usernameVariable: 'CFUSER', passwordVariable: 'CFPASS']]) {
+        sh "cf api ${env.PCF_API_ENDPOINT}"
+        sh "cf auth ${CFUSER} ${CFPASS}"
+        sh "cf target -o ${env.PCF_ORG} -s ${env.PHASE_ONE_PCF_SPACE}"
+        sh "cf push ${appName} -f manifest.jenkins.yml --hostname ${appName} -d ${env.PHASE_ONE_PCF_DOMAIN} --no-start"
+        sh "cf set-env ${appName} SPACE ${env.PHASE_ONE_PCF_SPACE}"
+        sh "cf set-env ${appName} DOMAIN ${env.PHASE_ONE_PCF_DOMAIN}"
+        sh "cf start ${appName}"
```

Comment:
We need to handle failures by deleting the app, or else we end up with hanging apps that never get deleted.

https://github.com/venicegeo/pz-gateway/blob/master/JenkinsFile#L108